flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -509,7 +536,8 @@ static ContainerLaunchContext createTaskExecutorContext(
 					appId,
 					new Path(taskManagerConfigFile.toURI()),
 					homeDirPath,
-					"").f1;
+					"",
+					null).f1;
```

Comment:
With my comment above we don't need to pass most of `null` default as here and else where.

Reply:
I think `null` means we always do not need to use pre-uploaded file for this local file. If you prefer add a new function and check not null outside of `setupSingleLocalResource`, i will do as you say.

Reply:
I think what @TisonKun means is if we use `setUpLocalResource()` and `registerPreloadedResource` as 2 different function. it is not needed to even pass in the null checker.